analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -241,6 +241,75 @@ public void Branching_LearnsObjectConstraint_Binary_Negated_VB(string expression
                 .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [DataTestMethod]
+        [DataRow("arg is object")]
+        [DataRow("arg is Exception")]
+        [DataRow("!!(arg is object)")]
+        [DataRow("!!(arg is Exception)")]
+        public void Branching_LearnsObjectConstraint_IsType_CS(string expression)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.IsType);
+            validator.ValidateTag("If", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("Else", x => x.Should().BeNull("it could be null or any other type"));
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x == null)
+                .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.NotNull));
+        }
+
+        [DataTestMethod]
+        [DataRow("!(arg is object)")]
+        [DataRow("!(arg is Exception)")]
+        [DataRow("!!!(arg is object)")]
+        [DataRow("!!!(arg is Exception)")]
+        public void Branching_LearnsObjectConstraint_IsType_Negated_CS(string expression)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.IsType);
+            validator.ValidateTag("If", x => x.Should().BeNull("it could be null or any other type"));
+            validator.ValidateTag("Else", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x == null)
+                .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.NotNull));
+        }
+
+        [TestMethod]
+        public void Branching_LearnsObjectConstraint_IsType_NoSymbol_DoesNotChangeState()
+        {
+            var validator = CreateIfElseEndValidatorCS("(object)arg is Exception", OperationKind.IsType);
```

Comment:
This is an FN. Isn't it? It applies in other cases too:
```cs
bool? b = true; int? i = 1;
!b is bool
-i is int
```

Reply:
In general, we don't propagate constraints inside nested operations (except conversion after review update on equals).

So yes, those cases would be missed.